lib/ruby/openai/client.rb

```diff
@@ -50,7 +50,7 @@ def self.post(path:, parameters: nil)
       HTTParty.post(
         uri(path: path),
         headers: headers,
```

Comment:
also the content type is wrong, but open-ai didn't care about it. maybe add an optional headers param. rename the `self.headers` to `self.defaut_headers`

then do `default_headers.merge(headers)`. 

also I feel it's not very scalable the way I did the conditional. the proper way is that each business class  calls itself `parameters.to_json`... Make it explicit

and in here, we only do : `body: parameters` without the .to_json call

Reply:
it would be a bit verbose, unfortunately, but cleaner IMO